doc/source/serve/tutorials/vllm-example.md

```diff
@@ -0,0 +1,51 @@
+---
+orphan: true
+---
+
+(serve-vllm-tutorial)=
+
+# Serve a Large Language Model with vLLM
+This example runs a Large Language model with Ray Serve using [vLLM](https://docs.vllm.ai/en/latest/), a popular open-source library for serving LLMs. It uses the [OpenAI Chat Completions API](https://platform.openai.com/docs/guides/text-generation/chat-completions-api) which can be used to easily integrate with other LLM tools. The example also demonstrates how to set-up multi-GPU serving with Ray Serve using placement groups.
+
+To run this example, install the following:
+
+```bash
+pip install "ray[serve]" requests vllm
+```
+
+This example uses the [NousResearch/Meta-Llama-3-8B-Instruct](https://huggingface.co/NousResearch/Meta-Llama-3-8B-Instruct) model. Save the following code to a file named llm.py.
+
+The Serve code is as follows:
+```{literalinclude} ../doc_code/vllm_openai_example.py
+:language: python
+:start-after: __serve_example_begin__
+:end-before: __serve_example_end__
+```
+
+Use `serve run llm:deployment model="NousResearch/Meta-Llama-3-8B-Instruct" tensor-parallel-size=2` to start the Serve application.
+
+:::{note}
+We are using tensor parallelism of 2 which means the model will be deployed to Ray Workers across 2 GPUs using placement groups in Ray Serve.
```

Comment:
nit: "tensor parallelism" is used here but "tensor-parallel-size" above, make them standard